web/src/main/webapp/heatMapArray/view.ts

```diff
@@ -0,0 +1,288 @@
+import {RemoteObjectView, CallbackCollector} from "../rpc";
+import {Dialog} from "../dialog";
+import {RangeInfo, TableView, Schema, IColumnDescription, isNumeric, isCategorical, DistinctStrings} from "../table";
+import {BasicColStats} from "../histogramBase";
+import {FullPage, Size, Resolution, IHtmlElement, ScrollBar, IScrollTarget} from "../ui";
+import {Pair, Triple} from "../util";
+import {ColumnAndRange} from "../histogramBase";
+import {HeatMapData} from "../heatmap";
+import {HeatMapArrayArgs, HeatMapArrayData} from "./types";
+import d3 = require('d3');
+
+class ColorMap {
```

Comment:
Maybe we can reuse this for the 2d heatmap too. Do you want to refactor that code to use it?